client/src/pages/BoardsPage/BoardsPage.js

```diff
@@ -120,19 +114,83 @@ const BoardsPage = _ => {
     }
   }
 
-  boardState.boardSync = () => {
-    const boardLength = JSON.parse(localStorage.getItem('boards'))
-
-    if (boardState.boards.length === boardLength) {
-      return false
+  boardState.handleSubmitBoard = e => {
+    console.log('submit board pressed')
+    e.preventDefault()
+    if(boardState.title){ // if title has an entry let the push happen
+      setBoardState({ ...boardState, isLoading: true}) // so we can disable the submit button after it is pressed once.
+
+      // get the user _id from the localStorage
+      boardState.getLocalStorageItem('user')
+        .then( ({_id}) => {
+
+          // for cleaner code, set the req.body to a variable
+          const payload = {
+            owner: _id,
+            title: boardState.title,
+            description: boardState.description
+          }
+
+          // post the created board in mongo
+          axios.post(`/api/boards`, payload)
+            .then( ({data}) => {
+              console.log('board created is...', data)
+              boardState.createTable(data._id)
+              boardState.getBoards()
+            })
+            .catch( e => console.error('error posting new board', e))
+        })
+        .catch( e => console.error('error getting user from storage', e))
     } else {
-      return true
+      boardState.logError({ message: 'Please enter a title'})
     }
   }
-  
+
+  boardState.createTable = boardId => {
+    console.log('running create table for board# ', boardId)
+
+    const newTable = {
+      board: boardId,
+      title: "New Table"
+    }
+
+    axios.post(`/api/tables`, newTable)
+    .then( r => {
+      console.log('successfully created table', r)
+
+      // generate tasks in the table
+      for( let k = 0; k<3 ; k++ ){
+        const taskPayload = {
+          table: r.data._id,
+          task: `New Task #${k}`
+        }
+
+        axios.post('/api/task', taskPayload)
+          .then( r => {
+            console.log('successfully added task to table')
+          })
+          .catch( e => console.error('error posting tasks to table', e))
+      } // end for loop generating tasks
+    })
+    .catch( e => console.error('error posting table to the database', e))
+  }
+
+  boardState.displayError = errors => errors.map((error, i) => <p key={i}>{error.message}</p>)
```

Comment:
This function came from the BottomBarPage. Putting it here because we moved the button from the bottom bar to the board page.